objects/src/testing/account.rs

```diff
@@ -73,11 +69,18 @@ impl<T: Rng> AccountBuilder<T> {
         self
     }
 
-    pub fn code<C: AsRef<str>>(mut self, code: C) -> Self {
-        self.code = code.as_ref().to_string();
+    pub fn code(mut self, account_code: AccountCode) -> Self {
+        self.code = Some(account_code);
         self
     }
 
+    /// Compiles [DEFAULT_ACCOUNT_CODE] into [AccountCode] and sets it.
+    pub fn default_code(self, assembler: Assembler) -> Self {
+        let default_account_code = AccountCode::compile(DEFAULT_ACCOUNT_CODE, assembler)
+            .expect("Default account code should compile.");
+        self.code(default_account_code)
+    }
```

Comment:
Not for this PR, but I find this method a bit counter-intuitive. Maybe instead of this we should have 2 `build()` methods? For example, something like this:
```rust
pub fn build(mut self) -> Result<(Account, Word), AccountBuilderError> {

}

pub fn build_default(mut self, assembler: Assembler) -> Result<(Account, Word), AccountBuilderError> {

}
```